pull.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @module  audio-generator
+ */
+'use strict';
+
+var Generator = require('./index');
+
+module.exports = function Stream (fn, opts) {
+	let generate = Generator(fn, opts);
+	let ended = false;
+
+	let stream = (end, cb) => {
+		if (end || ended) {
+			ended = true;
+			return cb && cb(end);
+		}
+
+		return cb(null, generate());
+	}
+
+	stream.abort = () => {ended = true}
```

Comment:
_infinite_ stream seem to be impossible to abort, unless by _Abortable_, therefore own code here. Otherwise it would be just `return infinite(generate);`